app/ts/background/background.ts

```diff
@@ -65,13 +65,16 @@ export const simulateGovernanceContractExecution = async (pendingTransaction: Pe
 		if (fourByte === undefined) return returnError('Could not identify the 4byte signature')
 		const explanation = FourByteExplanations[fourByte]
 		if ((explanation !== 'Submit vote' && explanation !== 'Cast vote') || pendingResults.data.simulatedAndVisualizedTransactions[0]?.events.length !== 1) return returnError('Could not identify the transaction as a vote')
-		if (pendingTransaction.transactionToSimulate.transaction.to === null) return  returnError('The transaction creates a contract instead of casting a vote')
+		
+		if (pendingTransaction.transactionToSimulate.transaction.to === null) return returnError('The transaction creates a contract instead of casting a vote')
 		const governanceContractInterface = new Interface(CompoundGovernanceAbi)
 		const params = governanceContractInterface.decodeFunctionData(explanation === 'Submit vote' ? 'submitVote' : 'castVote', dataStringWith0xStart(pendingTransaction.transactionToSimulate.transaction.input))
-		const contractExecutionResult = await simulateCompoundGovernanceExecution(ethereum, pendingTransaction.transactionToSimulate.transaction.to, params[0])
+		const addr = await identifyAddress(ethereum, userAddressBook, pendingTransaction.transactionToSimulate.transaction.to)
+		if (!('abi' in addr) || addr.abi === undefined) return { error: { type: 'MissingAbi' as const, message: 'ABi for the governance contract is missing', addressBookEntry: addr } }
+		const contractExecutionResult = await simulateCompoundGovernanceExecution(ethereum, addr, params[0])
 		if (contractExecutionResult === undefined) return returnError('Failed to simulate governacne execution')
 		const parentBlock = await ethereum.getBlock()
-		if (parentBlock.baseFeePerGas === undefined) throw new Error('cannot build simulation from legacy block')
+		if (parentBlock.baseFeePerGas === undefined) return returnError('cannot build simulation from legacy block')
```

Comment:
I wonder if, at this point, we could just remove support for pre-London blocks entirely?  Is there anything we do in Interceptor that actually needs to support them?

Reply:
> I wonder if, at this point, we could just remove support for pre-London blocks entirely? Is there anything we do in Interceptor that actually needs to support them?

a DAPP can Query for an old block header/block I think. Some dapp was querying for block 1 for example (cannot remember which)